simulation/app.py

```diff
@@ -0,0 +1,621 @@
+            Length in which instances inside container could move.
+
+    Returns
+    -------
+    Based on selected event return True or False for dcc.Interval object's attribute
+    'disabled'. If event is Stop, Reset returns True, if Start, Continue returns
+    False.
+    """
+
+    ctx = dash.callback_context
+
+    if not ctx.triggered:
+        button_id = 'No clicks yet'
+    else:
+        button_id = ctx.triggered[0]['prop_id'].split('.')[0]
+
+    if button_id == 'start-button':
+        box.population = population_amount
+        box.move_distance_length = move_length
+        box.initial_set_up(number_of_susceptible=susceptible_amount,
+                           number_of_infected=infected_amount,
+                           number_of_recovered=recovered_amount,
+                           number_of_dead=dead_amount,
+                           infection_probability=infection_probability,
+                           recover_probability=recover_probability,
+                           dead_probability=dead_probability,
+                           infection_range=infection_range)
+        old_amounts['susceptible'] = susceptible_amount
+        old_amounts['recovered'] = recovered_amount
+        old_amounts['dead'] = dead_amount
+        return False
+
+    if button_id == 'stop-button' or box.count_infected() <= 0:
+        return True
+
+    if button_id == 'continue-button':
+        return False
+
+    if button_id == 'reset-button':
+        d['condition'] = []
+        d['time'] = []
+        d['value'] = []
+
+        box.population = 0
+        box.initial_set_up(0, 0, 0, 0, 0, 0, 0, 0)
+        box.object_list = []
+
+        return True
+
+    return True
+
+
+@app.callback(Output('live-population', 'figure'),
+              [Input('interval-component', 'n_intervals'),
+               Input('simulation-data', 'data')])
+def update_graph_live(n_intervals: int, json_data: json) -> 'plotly_express.line':
+    """
+    Function handling scatter graph for each individual instance move in
+    grid.
+
+    Parameters
+    ----------
+    n_intervals: int, required
+        Number of passed intervals from begin of simulation. Generated
+        with dash object dcc.Interval.
+    json_data: json, required
+        JSON contain data of simulation.
+
+    Returns
+    -------
+    Return plolty express line chart with information of current amount of
+    each group in simulation.
+    """
+
+    data = json.loads(json_data)
+
+    d['time'].append(n_intervals)
+    d['value'].append(data['groups']['susceptible'])
+    d['condition'].append('susceptible')
+    d['time'].append(n_intervals)
+    d['value'].append(data['groups']['infected'])
+    d['condition'].append('infected')
+    d['time'].append(n_intervals)
+    d['value'].append(data['groups']['recovered'])
+    d['condition'].append('recovered')
+    d['time'].append(n_intervals)
+    d['value'].append(data['groups']['dead'])
+    d['condition'].append('dead')
```

Comment:
Maybe there is a method to preserve last content of the graph ans draw only next points? I do not know, maybe there is something better than appending each time and draw entire plot from scratch. Need to search.